test/functional/public_uploads_controller_test.rb

```diff
@@ -0,0 +1,24 @@
+require "test_helper"
+
+class PublicUploadsControllerTest < ActionController::TestCase
+  test "redirects asset requests that aren't made via the asset host" do
+    Plek.any_instance.stubs(:public_asset_host).returns('http://asset-host.com')
+    request.host = 'not-asset-host.com'
+
+    get :show, params: { path: 'asset', format: 'txt' }
+
+    assert_redirected_to 'http://asset-host.com/government/uploads/asset.txt'
+  end
+
+  test 'does not redirect asset requests that are made via the asset host' do
+    asset_filesystem_path = File.join(Whitehall.clean_uploads_root, 'asset.txt')
+    FileUtils.touch(asset_filesystem_path)
+
+    Plek.any_instance.stubs(:public_asset_host).returns('http://asset-host.com')
```

Comment:
Minor: This line appears in both tests and could be extracted into a `setup` block.

Reply:
Good spot. I've moved this to a `setup` block.